src/vs/editor/contrib/stickyScroll/browser/stickyScrollWidget.ts

```diff
@@ -40,13 +57,106 @@ export class StickyScrollWidget extends Disposable implements IOverlayWidget {
 		this.lineNumbers = [];
 		this.lastLineRelativePosition = 0;
 
+		this.hoverOnLine = -1;
+		this.positionOfDefinitionToJumpTo = { uri: {}, range: {} } as Location;
+		this.lastChildDecorated = undefined;
+		this.stickyPositionProjectedOnEditor = new Position(-1, -1);
+
 		this.lineHeight = this._editor.getOption(EditorOption.lineHeight);
 		this._register(this._editor.onDidChangeConfiguration(e => {
 			if (e.hasChanged(EditorOption.lineHeight)) {
 				this.lineHeight = this._editor.getOption(EditorOption.lineHeight);
 			}
 		}));
+		this._register(this.updateLinkGesture());
+	}
+
+	private updateLinkGesture(): IDisposable {
+		const gesture = this.linkGestureStore.add(new ClickLinkGesture(this._editor));
+		const cancellationToken = new CancellationTokenSource();
+		const sessionStore = new DisposableStore();
+		sessionStore.add(cancellationToken);
+		this.linkGestureStore.add(sessionStore);
+		this.linkGestureStore.add(gesture.onMouseMoveOrRelevantKeyDown(([mouseEvent, _keyboardEvent]) => {
+			if (!this._editor.hasModel()) {
+				return;
+			}
+			if (!mouseEvent.hasTriggerModifier) {
```

Comment:
we can merge these statements and can return on first match.

Reply:
That's right, thanks for the tip